tests/p2p/test_server.py

```diff
@@ -44,8 +44,15 @@ def get_open_port():
 INITIATOR_REMOTE = Node(INITIATOR_PUBKEY, INITIATOR_ADDRESS)
 
 
-class MockPeerPool:
+class MockPeerPool():
     is_full = False
+    connected_nodes = {}
+
+    def is_valid_connection_candidate(self, node):
```

Comment:
@williambannas can you look into adding tests to cover the functionality of the PR